### PR TITLE
improve race condition handling

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

### Description

This PR further improves the race condition handling in RM, especially in handling requests after reply was sent.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by nvidianz — APPROVED** [review]
Tested and no issues.

File: nvflare/apis/utils/reliable_message.py [hunk 1]
```diff
@@ -295,15 +295,35 @@ def _receive_request(cls, topic: str, request: Shareable, fl_ctx: FLContext):
...
+                # no standing process for this request
+                # further check whether this request was already completed
+                if cls._req_completed.get(tx_id):
+                    # this request was already completed!
+                    cls.debug(fl_ctx, "Completed tx_id received")
+                    return _status_reply(STATUS_DUP_REQUEST)
+
+            if not receiver:
+                # this is a valid new request
+                receiver = cls._get_or_create_receiver(rm_topic, request, handler_f)
```
Comment (YuanTingHsieh) [inline nvflare/apis/utils/reliable_message.py:315]: this part can be put inside previous block since they are under same condition "if not receiver"

**Review by YuanTingHsieh — COMMENTED** [review]
LGTM, just one minor comment

File: nvflare/apis/utils/reliable_message.py (hunk 1 above)
Comment (yanchengnv) [reply nvflare/apis/utils/reliable_message.py:315]: I kept them separate because they are for different purposes.

**Pull request merged**